pkg/volume/iscsi/iscsi_util_test.go

```diff
@@ -275,43 +282,48 @@ func TestClonedIfaceShowError(t *testing.T) {
 	plugin := plugins[0]
 	fakeMounter := iscsiDiskMounter{
 		iscsiDisk: &iscsiDisk{
-			Iface:  "192.168.1.10:pv0001",
+			Iface:  TestIface,
 			plugin: plugin.(*iscsiPlugin)},
 		exec: fakeExec,
 	}
-	cloneIface(fakeMounter)
-	if cmdCount != 1 {
-		t.Errorf("expected 1 CombinedOutput() calls, got %d", cmdCount)
+	err := cloneIface(fakeMounter)
+	if err == nil {
+		t.Errorf("expect to receive error, nil received")
+	}
+	if fakeExec.CommandCalls != len(scripts) {
+		t.Errorf("expected 1 CombinedOutput() calls, got %d", fakeExec.CommandCalls)
 	}
 
 }
 
 func TestClonedIfaceUpdateError(t *testing.T) {
-	cmdCount := 0
-	fakeExec := mount.NewFakeExec(func(cmd string, args ...string) ([]byte, error) {
-		cmdCount++
-		if cmd != "iscsiadm" {
-			t.Errorf("iscsiadm command expected, got %q", cmd)
-		}
-		switch cmdCount {
-		case 1:
-			// iscsiadm -m iface -I <iface> -o show
-			return []byte("iface.ipaddress = <empty>\niface.transport_name = tcp\niface.initiatorname = <empty>\n"), nil
-
-		case 2:
-			// iscsiadm -m iface -I <newIface> -o new
-			return []byte("New interface 192.168.1.10:pv0001 added"), nil
-		case 3:
-			// iscsiadm -m iface -I <newIface> -o update -n <key> -v <val>
-			return []byte(""), nil
-		case 4:
-			return []byte(""), errors.New("test error")
-		case 5:
-			// iscsiadm -m iface -I <newIface> -o delete
-			return []byte(""), nil
-		}
-		return nil, fmt.Errorf("Unexpected exec call nr %d: %s", cmdCount, cmd)
-	})
+	fakeExec := &testingexec.FakeExec{}
+	scripts := []volumetest.CommandScript{
+		{
+			Cmd:  "iscsiadm",
+			Args: []string{"-m", "iface", "-I", "", "-o", "show"},
+			// Output: "iface.ipaddress = <empty>\niface.transport_name = tcp\niface.initiatorname = <empty>\n"
```

Comment:
where is the output actually set?

Reply:
I put these comments in to carry-over what was there before. I didn't need it to get the tests to pass. If we needed to set the output, it would be set here.

Looking closer at the `cloneIface` function, it looks like it just goes on just fine if the command output is an empty string. I will run this test again with the output set and make sure it still works okay. It's only the output from `iscsiadm show` that seems to get used -- so I can add in.